src/server.ts

```diff
@@ -81,66 +83,166 @@ export async function buildServer (ctx: ViteBuildContext) {
+async function generateBuildManifest (ctx: ViteBuildContext) {
+  const serverDist = resolve(ctx.nuxt.options.buildDir, 'dist/server')
+  const clientDist = resolve(ctx.nuxt.options.buildDir, 'dist/client')
+  const r = (...args: string[]): string => resolve(serverDist, ...args)
+
+  const publicPath = '/_nuxt/'
+  const viteClientManifest = await readJSON(join(clientDist, 'manifest.json'))
+
+  function getModuleIds ([, value]: [string, any]) {
+    if (!value) {
+      return []
+    }
+    return [value.file, ...value.css || []]
+      .filter(i => !i.endsWith('.js') || i.match(/-legacy\./)) // only use legacy build
+  }
+
+  const asyncEntries = uniq(
+    Object.entries(viteClientManifest)
+      .filter((i: any) => i[1].isDynamicEntry)
+      .flatMap(getModuleIds)
+  ).filter(i => i)
+  const initialEntries = uniq(
+    Object.entries(viteClientManifest)
+      .filter((i: any) => !i[1].isDynamicEntry)
+      .flatMap(getModuleIds)
+  ).filter(i => i)
+  const initialJs = initialEntries.filter(i => i.endsWith('.js'))
+  const initialAssets = initialEntries.filter(i => !i.endsWith('.js'))
+
+  // search for polyfill file, we don't include it in the client entry
+  const polyfillName = initialEntries.find(i => i.startsWith('polyfills-legacy.'))
+
+  // @vitejs/plugin-legacy uses SystemJS which need to call `System.import` to load modules
+  const clientEntryCode = initialJs
+    .filter(i => !i.startsWith('polyfills-legacy.'))
+    .map(i => `System.import("${publicPath}${i}");`)
+    .join('\n')
+  const clientEntryHash = createHash('sha256')
+    .update(clientEntryCode)
+    .digest('hex')
+    .substr(0, 8)
+  const clientEntryName = 'entry.' + clientEntryHash + '.js'
 
-  watcher.on('change', () => build())
+  const clientManifest = {
+    publicPath,
+    all: uniq([
+      polyfillName,
+      clientEntryName,
+      ...Object.entries(viteClientManifest)
+        .flatMap(getModuleIds)
+    ]).filter(i => i),
+    initial: [
+      polyfillName,
+      clientEntryName,
+      ...initialAssets
+    ],
+    async: [
+      // we move initial entries to the client entry
+      ...initialJs,
+      ...asyncEntries
+    ],
+    modules: {},
+    assetsMapping: {}
+  }
+  const serverManifest = {
+    entry: 'server.js',
+    files: {
+      'server.js': 'server.js',
+      ...Object.fromEntries(Object.entries(viteClientManifest).map((i: any) => [i[0], i[1].file]))
+    },
+    maps: {}
+  }
+
+  await writeFile(join(clientDist, clientEntryName), clientEntryCode, 'utf-8')
+  await writeJSON(r('client.manifest.json'), clientManifest, { spaces: 2 })
+  await writeJSON(r('server.manifest.json'), serverManifest, { spaces: 2 })
+}
+
+// stub manifest on dev
+async function stubManifest (ctx: ViteBuildContext) {
+  const serverDist = resolve(ctx.nuxt.options.buildDir, 'dist/server')
+  const r = (...args: string[]): string => resolve(serverDist, ...args)
+
+  await writeJSON(r('client.manifest.json'), {
```

Comment:
As I tested vite with Nitro, I think we should also create `client.manifest.mjs`, because Nitro looks for `client.manifest.mjs`

Reply:
/cc @danielroe do you have an example of its format that I can reference to?

Reply:
The format is just (effectively):
```js
await writeJSON(r('client.manifest.json'), clientManifest, { spaces: 2 })
await writeFile(r('client.manifest.mjs'), `export default ${JSON.stringify(clientManifest, { spaces: 2 })}`)
```